frontend/src/components/input/ListInput.vue

```diff
@@ -0,0 +1,145 @@
+<script lang="ts" setup>
+import { ref, computed, watch, nextTick, onMounted } from "vue"
+import Checkbox from "@/components/input/CheckboxInput.vue"
+import InputField, { LabelPosition } from "@/components/input/InputField.vue"
+import TextButton from "@/components/input/TextButton.vue"
+
+const props = defineProps<{
+  label: string
+  modelValue: string[]
+}>()
+
+const emit = defineEmits<{
+  "update:modelValue": [value: string[]]
+}>()
+
+const editMode = ref(true)
+const sortAlphabetically = ref(false)
+const localModelValue = ref(props.modelValue ?? [])
+
+const textAreaInput = computed({
+  get: () => (props.modelValue ? props.modelValue.join("\n") : ""), // Join array with newlines for textarea
+  set: (newValues: string) => {
+    // Split the text by newline, trim each line, filter out empty lines, and set back into the array
+    localModelValue.value = newValues
+      .split("\n")
+      .map((listitem) => listitem.trim())
+      .filter((listitem) => listitem !== "")
+  },
+})
+
+const addList = () => {
+  // sort alphabetically if option set
+  if (sortAlphabetically.value && localModelValue.value) {
+    localModelValue.value = localModelValue.value.sort((a: string, b: string) =>
+      a.localeCompare(b),
+    )
+  }
+  emit("update:modelValue", [...new Set(localModelValue.value)] as string[]) //remove duplicates
+  if (!!localModelValue.value?.length) editMode.value = false
+  sortAlphabetically.value = false
+}
+const cancelEdit = () => {
+  if (!!localModelValue.value?.length) editMode.value = false
+}
+
+const toggleEditMode = () => {
+  editMode.value = !editMode.value
+}
+
+const adjustTextareaHeight = (textarea: HTMLTextAreaElement | null) => {
+  if (textarea) {
+    textarea.style.height = "auto" // Reset height first to recalculate based on content
+    textarea.style.height = `${textarea.scrollHeight}px ` // Set the height to match content
+  }
+}
+
+// Watch the `textAreaInput` value to adjust the height when content changes
+watch(textAreaInput, async () => {
+  await nextTick() // Wait for DOM render
+  const textarea = document.querySelector("textarea")
+  adjustTextareaHeight(textarea as HTMLTextAreaElement)
+})
+
+onMounted(() => {
+  // When data, show display mode. When empty, show edit mode
+  editMode.value = !props.modelValue.length
+})
+</script>
+
+<template>
+  <div>
+    <!-- Edit mode -->
+    <div v-if="editMode" class="flex flex-col gap-24">
+      <div class="flex flex-col gap-8">
```

Comment:
```suggestion
      <div class="flex flex-col">
```
In the design we only have a gap of 4px which the label already has as `mb`.
<img width="314" alt="image" src="https://github.com/user-attachments/assets/49770042-f144-4232-9cc0-923d3eba84fc">